### PR TITLE
Fixed PMD violation God Class for RealInternalStore

### PR DESCRIPTION
**Pull request by KevCron**
Resolved PMD issues "God Class" by moving some methods to `StoreUtil`. Seemed (at least to me) like they should logically be in there any way.

### CODE REVIEW

**Review by brianPlummer — APPROVED** [review]

**Comment by brianPlummer** [comment]
Looks great!!! good job Kevin!!! thank you so much!!

**Comment by pavlospt** [comment]
@KevCron Nice one! If you use StoreUtil inside `initMemCache()` and `initFlightRequets()` you can also avoid the final PMD warning suppression :D  

`@SuppressWarnings("PMD.AvoidReassigningParameters")`

**Comment by KevCron** [comment]
@pavlospt I believe the final PMD warning comes from line [67](https://github.com/NYTimes/Store/blob/develop/store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java#L70), where `memoryPolicy` is assigned a value, let me know if I'm mistaken. Are you saying I should create a default `memoryPolicy` in `StoreUtil`, and not use the parameter?

**Comment by KevCron** [comment]
Also @brianPlummer I don't have write access, so I can't merge the request. You can just hit merge if you think everything looks good. And you're welcome! Let me know if there's anything else I can focus on, I want to contribute to this repo more.

**Comment by KevCron** [comment]
Fixed last PMD (Parameter reassignment) violation as well.

**Review by digitalbuddha — COMMENTED** [review]
I'm not crazy about it. It would be nice to split into better objects like StoreInitizalizer. I'll over feedback in am

**Comment by KevCron** [comment]
So here's what I'm thinking

- We move `initMemCache` and `initFlightRequests` to a `StoreInitializer` class
- `clearPersister` should stay in `StoreUtil` (to me, it seems like a utility)
- `getCacheTTL`, getCacheSize, and `getCacheTTLTimeUnit` in `StoreInitializer` as it seems to me where they belong

Thoughts?

**Comment by digitalbuddha** [comment]
That sounds great! will merge after changes

**Comment by pavlospt** [comment]
@KevCron @digitalbuddha You can also make a new class, named `StoreDefaults` and add `getCacheTTL`, `getCacheSize` and `getCacheTTLTimeUnit` there, since they are not mutable. Could be of use in future as well, if there are to be more default values like these.

File: store/src/main/java/com/nytimes/android/external/store/base/impl/StoreInitializer.java
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/StoreInitializer.java]: Let's rename this to `CacheFactory` and have methods  `createCache`, `createInflighter`

File: store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java:64]: This looks so much better!!

**Review by digitalbuddha — APPROVED** [review]

**Comment by digitalbuddha** [comment]
Thanks for following through and landing this! Big help

**Pull request merged**
